### ISSUE TITLE
Playback of streaming audio content

### ISSUE THREAD

**Issue by disansk** [sagiadinos/garlic-player#37]
**Describe the bug**
The player does not play streaming audio content

**To Reproduce**
Add an audio stream to the playlist:
<audio src="http://audio.test.local:80/gorvolna" soundLevel="20%" dur="30s" />
This is link to server icecast with codec MPEG Audio layer 1/2 (mpga)

The player skips the task

In event_log.xml I see an entry:
<event><eventType>error</eventType><eventTime>2023-03-02T17:18:19+07:00</eventTime><eventSource>ContentManager</eventSource><eventName>FETCH_FAILED</eventName><metadata><meta name="resourceURI" content="http://audio.test.local/gorvolna"/><meta name="errorMessage" content="Error transferring http://audio.test.local/gorvolna - server replied: Bad Request"/><meta name="transferLength" content="0"/></metadata></event>

This link plays in vlc player and in any browser

**Additional context**
An additional question concerns the playback of a video stream. Can garlic-player play a video stream, and if so, which one?

**Comment by disansk** [comment]
incorrectly inserted xml string:
`<event><eventType>error</eventType><eventTime>2023-03-02T17:18:19+07:00</eventTime><eventSource>ContentManager</eventSource><eventName>FETCH_FAILED</eventName><metadata><meta name="resourceURI" content="http://audio.test.local/gorvolna"/><meta name="errorMessage" content="Error transferring http://audio.test.local/gorvolna - server replied: Bad Request"/><meta name="transferLength" content="0"/></metadata></event>`

**Comment by sagiadinos** [comment]
Thank you for report.

Currently, the garlic-player does not support video/audio-steams direct. 

As a workaround: You can use them on a website or as a so-called [Digital Signage Widget.](https://smil-control.com/magazine/digital-signage-widgets/)

**Comment by disansk** [comment]
> Thank you for report.
> 
> Currently, the garlic-player does not support video/audio-steams direct.
> 
> As a workaround: You can use them on a website or as a so-called [Digital Signage Widget.](https://smil-control.com/magazine/digital-signage-widgets/)

Thank you! Let's consider this a feature request

**Comment by disansk** [comment]
Hello!
Is the feature planned to be implemented?
Thank you for your work!

**Comment by sagiadinos** [comment]
Yes, but it will take some rework in the lib and cache mechanism. Honestly not the biggest priority at the moment as I want to rewrite the lib to get rid of Qt Lib.

[Streaming.wgt.zip](https://github.com/user-attachments/files/31221312/Streaming.wgt.zip)

If you like you can try the attached widget (Streaming.zp). JUst unpack to wgt before use. I had this widget some times ago, in my pipeline and uploaded yesterday to https://github.com/garlic-signage/garlic-widgets

There are no docs yet, but they will come soon.

If you want to know more about html5 digital signage widgets you find here it at the [garlic widget page](https://garlic-signage.com/garlic-widgets/).
